server/testnet.go

```diff
@@ -0,0 +1,174 @@
+package server
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	gc "github.com/cosmos/cosmos-sdk/server/config"
+
+	"github.com/cosmos/cosmos-sdk/wire"
+	"github.com/spf13/viper"
+	cfg "github.com/tendermint/tendermint/config"
+	cmn "github.com/tendermint/tmlibs/common"
+)
+
+var (
+	nodeDirPrefix = "node-dir-prefix"
+	nValidators   = "v"
+	outputDir     = "o"
+
+	startingIPAddress = "starting-ip-address"
+)
+
+const nodeDirPerm = 0755
+
+// get cmd to initialize all files for tendermint testnet and application
+func TestnetFilesCmd(ctx *Context, cdc *wire.Codec, appInit AppInit) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "testnet",
+		Short: "Initialize files for a Gaiad testnet",
+		Long: `testnet will create "v" number of directories and populate each with
+necessary files (private validator, genesis, config, etc.).
+
+Note, strict routability for addresses is turned off in the config file.
+
+Example:
+
+	gaiad testnet --v 4 --o ./output --starting-ip-address 192.168.10.2
+	`,
+		RunE: func(_ *cobra.Command, _ []string) error {
+			config := ctx.Config
+			err := testnetWithConfig(config, ctx, cdc, appInit)
+			return err
+		},
+	}
+	cmd.Flags().Int(nValidators, 4,
+		"Number of validators to initialize the testnet with")
+	cmd.Flags().String("o", "./mytestnet",
```

Comment:
I think you meant to use `outputDir` instead of `"o"`

Reply:
Interestingly, this is the input parameter in Tendermint and I wanted to keep the two compatible. I can get rid of that idea, since the code changed so much. I'm renaming it to `--outputdir`.